core/src/main/scala/kafka/api/ApiVersion.scala

```diff
@@ -24,16 +24,24 @@ package kafka.api
  *
  * Note that the ID we initialize for each version is important.
  * We consider a version newer than another, if it has a higher ID (to avoid depending on lexicographic order)
+ * 
+ * If there is a draft protocol version between protocols of two official releases. The suffix "-DV#" will be added to
+ * help users who are running on trunk upgrade. For example:
+ * 1. Kafka 0.9.0 is released
+ * 2. After that some protocol change are made and will be released in 0.10.0. The version will be named 0.10.0-DV0
+ *    (DV stands for draft version)
+ * 3. When Kafka 0.10.0 is released. The official version 0.10.0 will be the same as the last draft version.
  */
 object ApiVersion {
   // This implicit is necessary due to: https://issues.scala-lang.org/browse/SI-8541
   implicit def orderingByVersion[A <: ApiVersion]: Ordering[A] = Ordering.by(_.id)
 
   private val versionNameMap = Map(
-    "0.8.0" -> KAFKA_080,
-    "0.8.1" -> KAFKA_081,
-    "0.8.2" -> KAFKA_082,
-    "0.9.0" -> KAFKA_090
+    "0.8.0" -> KAFKA_0_8_0,
+    "0.8.1" -> KAFKA_0_8_1,
+    "0.8.2" -> KAFKA_0_8_2,
+    "0.9.0" -> KAFKA_0_9_0,
+    "0.10.0-DV0" -> KAFKA_0_10_0_DV0
```

Comment:
In addition to have the internal versions, we probably should always have an "0.10.0" version in trunk so that it's in a releasable state. I was thinking that we always have "0.10.0" map to the last case object. In this case, both "0.10.0" and "0.10.0-DV0" will be pointing to KAFKA_0_10_0_DV0. When we add KAFKA_0_10_0_DV1, "0.10.0" will be pointing to KAFKA_0_10_0_DV1 and we will add "0.10.0-DV1", but leave "0.10.0-DV0" unchanged. In the code, we can just reference the first internal version in which a format change is introduced. Then, for people deploy from trunk, they can use the internal version. For people who want to try trunk, they can use "0.10.0".

Also, would it be better to use IV (internal version) instead of DV?

Finally, it may make sense to make the next major release 0.10 instead of 0.9.1 since we will be including KStream. Do you want to poll the mailing list to see if people are ok with this?